### ISSUE TITLE
changes not done

### ISSUE THREAD

**Issue by Ernestopheles** [microsoft/vscode-copilot-release#2261]
Type: <b>Bug</b>

in the file there are no changes by copilot; happened quite often...

Extension version: 0.23.2024102903
VS Code version: Code 1.95.1 (65edc4939843c90c34d61f4ce11704f09d3e5cb6, 2024-10-31T05:14:54.222Z)
OS version: Linux x64 6.8.0-48-generic snap
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|13th Gen Intel(R) Core(TM) i7-1355U (12 x 2565)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software<br>webgpu: disabled_off<br>webnn: unavailable_software|
|Load (avg)|2, 2, 2|
|Memory (System)|31.09GB (20.67GB free)|
|Process Argv|--no-sandbox --force-user-env --crash-reporter-id 5e9f53b6-f610-4ec0-8d96-c977e405793c|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|plasma|
|XDG_CURRENT_DESKTOP|KDE|
|XDG_SESSION_DESKTOP|KDE|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
vscaac:30438847
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
pythonnoceb:30805159
asynctok:30898717
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
0ee40948:31013168
dvdeprecation:31068756
dwnewjupytercf:31046870
impr_priority:31102340
nativerepl2:31139839
refactort:31108082
pythonrstrctxt:31112756
cf971741:31144450
iacca1:31171482
notype1cf:31157160
5fd0e150:31155592
dwcopilot:31170013

```

</details>

**Comment by TylerLeonhardt** [comment]
What Copilot entrypoint are you using?
* Chat
* Editor Inline Chat
* Copilot Edits
* Quick Chat
? 

A screenshot would also be helpful.

**Comment by Ernestopheles** [comment]
Sorry...

Copilot edit
Copilot writes that it has made changes to a file - but there is no diff; accept/discard is shown nethertheless

**Comment by joyceerhl** [comment]
Can you please share reproducible steps, i.e. the files that you attach to the working set and your user query?

**Comment by sean-mcmanus** [comment]
I repro this with the latest Insiders but not non-Insiders. I have some repro details at https://github.com/microsoft/vscode-copilot-release/issues/2314 . I can provide more repro details if you want. My issue appears 100% reproducible.

**Comment by joyceerhl** [comment]
@sean-mcmanus yes, can you please provide the prompt you are sending and the file that you're editing? Feel free to email it to me if you don't want to upload it here

**Comment by sean-mcmanus** [comment]
Repro steps (you might be able to simplify the repro, I haven't tried):

1. git clone https://github.com/microsoft/vscode-cpptools.git . (The "main" branch).
2. Open the vscode-cpptools/Extension folder.
3. Open the CHANGELOG.md file, i.e. https://github.com/microsoft/vscode-cpptools/blob/main/Extension/CHANGELOG.md
4. Use the Copilot Edits view.
5. Paste one of these prompts:
```
Review this changelog: * A potential fix for a crash during process shutdown (in `uv_run`). [#12668](https://github.com/microsoft/vscode-cpptools/issues/12668)
* Fix a performance issue where some LSP requests would delay other LSP requests. [#12905](https://github.com/microsoft/vscode-cpptools/issues/12905)
* A potential fix for a crash in cpptools (in `report_intellisense_results`).
* Fix a random deadlock with `compiler_info::find_or_create`.
* Fix a random deadlock with `handle_edits`.
* Other internal fixes.
```
or
```
Review this changelog: * Fix some new bug to test chat.
* A potential fix for a crash during process shutdown (in `uv_run`). [#12668](https://github.com/microsoft/vscode-cpptools/issues/12668)
* Fix a performance issue where some LSP requests would delay other LSP requests. [#12905](https://github.com/microsoft/vscode-cpptools/issues/12905)
* A potential fix for a crash in cpptools (in `report_intellisense_results`).
* Fix a random deadlock with `compiler_info::find_or_create`.
* Fix a random deadlock with `handle_edits`.
* Other internal fixes.
```

Result: In both cases, the response sounds like an edit was intended, but there is no edit to the CHANGELOG.md file.
![Image](https://github.com/user-attachments/assets/7d668d39-321b-46c5-854c-7220dd2624ca)
or
![Image](https://github.com/user-attachments/assets/cc49b648-1c92-486a-9433-0c6596ee0b95)

However, the good news is that I found a workaround by switching to the Chat view instead of the Copilot Edits view (I was clicking the icons to try to clear the chat history and didn't realize I had switched to a new mode).

**Comment by joyceerhl** [comment]
I can repro, and it looks like the file is simply too long.

**Comment by sudanakunyung** [comment]
![Image](https://github.com/user-attachments/assets/569dae02-0c3a-487e-a58f-f8c7b628e78a)
maybe you can try

**Comment by they-cloned-me** [comment]
Same issue, but only when I use `Gemini 2.5 Pro (Preview)`

<img width="363" alt="Image" src="https://github.com/user-attachments/assets/9bb21ee4-a41a-4e30-ab6d-f0abc31a9cc1" />

**Comment by joyceerhl** [comment]
@they-cloned-me can you please share a prompt and file / set of files where this happens? We can't investigate otherwise

**Comment by they-cloned-me** [comment]
>[!WARNING]
>The silly file and prompt are for reporting this only

For my case, it’s just a simple css file, and the prompt could be as basic as “**Remove code comments**”
So I’m not sure it’s really about file length.

To reiterate: this only happens with `Gemini 2.5 Pro (Preview)`.
I’ve been using all the other models for long and complex tasks (especially Claude 3.7), but when I switch to Gemini, oopsie

TLDR; I’ve not be able to successfully use `Gemini 2.5 Pro (Preview)` because of this.

```scss
*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html, body {
  overflow: hidden;
  height: 100%;
  width: 100%;
  font-family: 'Inter', system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, sans-serif;
  background-color: black;
  color: white;
  scroll-behavior: smooth;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-rendering: optimizeLegibility;
}

/* Custom selection color */
::selection {
  background-color: rgba(255, 64, 160, 0.5); /* Pink from left panel */
  color: white;
}

/* Remove default focus outline and add custom one */
:focus {
  outline: none;
}

:focus-visible {
  outline: 2px solid #30b0ff;
  outline-offset: 2px;
}
```

It sucks because I don’t have any error or warning logs in the editor

**Comment by they-cloned-me** [comment]
cc @joyceerhl

**Comment by franklin-tina** [comment]
forgive the typos @joyceerhl 

https://github.com/user-attachments/assets/5af88114-45e4-4753-ae7b-34156c5e2d4f

**Comment by borisalmonacid** [comment]
same error

**Comment by joyceerhl** [comment]
@rebase what 503 error are you getting, can you please specify the model you're using?

**Comment by rebase** [comment]
@joyceerhl It’s working fine for now. The 503 error occurred when I was using Claude Sonnet 4 in Agent mode.
